client/src/api/common/data.ts

```diff
@@ -0,0 +1,69 @@
+import { useState } from "react";
```

Comment:
Remind me, did we discuss putting this in `common` vs `v3`? Right now it's V3-specific, but I understand that future versions of the bike may also use the same format.

Reply:
I'm not sure if that's something we discussed. I agree it is currently only for `v3` but I anticipate that this will be what we're going with in the future. It seems like keeping it in common and moving it if we do decide on changing the types would be the same amount of work as moving it to `v3` now and then maybe moving it back to common depending on `v4`.

Reply:
I agree with @haotongwang. Because it is relatively to change later we can easily keep it in common and change it if necessary for  `v4`.